crates/bevy_ecs/src/system/system_piping.rs

```diff
@@ -73,29 +73,24 @@ where
 /// This trait is blanket implemented for all system pairs that fulfill the type requirements.
 ///
 /// See [`PipeSystem`].
-pub trait IntoPipeSystem<ParamA, Payload, SystemB, ParamB, Out>:
-    IntoSystem<(), Payload, ParamA> + Sized
-where
-    SystemB: IntoSystem<Payload, Out, ParamB>,
-{
+pub trait IntoPipeSystem<In, Payload, Marker>: IntoSystem<In, Payload, Marker> + Sized {
     /// Pass the output of this system `A` into a second system `B`, creating a new compound system.
-    fn pipe(self, system: SystemB) -> PipeSystem<Self::System, SystemB::System>;
-}
-
-impl<SystemA, ParamA, Payload, SystemB, ParamB, Out>
-    IntoPipeSystem<ParamA, Payload, SystemB, ParamB, Out> for SystemA
-where
-    SystemA: IntoSystem<(), Payload, ParamA>,
-    SystemB: IntoSystem<Payload, Out, ParamB>,
-{
-    fn pipe(self, system: SystemB) -> PipeSystem<SystemA::System, SystemB::System> {
+    fn pipe<B, Out, MarkerB>(self, system: B) -> PipeSystem<Self::System, B::System>
+    where
+        B: IntoSystem<Payload, Out, MarkerB>,
+    {
```

Comment:
I wonder if we could just implement this as a default function on `IntoSystem` and do away with this trait.

Reply:
Good idea, I'll try that.